log/wrapper.go

```diff
@@ -180,29 +195,39 @@ func TestWrapper(tb testing.TB) Wrapper {
 	}
 }
 
+// ZapWrapperArgs defines the args used in ZapWrapper.
+type ZapWrapperArgs struct {
+	Level   Level
+	KVPairs map[string]interface{}
+}
+
 // ZapWrapper wraps zap log package into a Wrapper.
-func ZapWrapper(level Level) Wrapper {
-	if level == NopLevel {
+func ZapWrapper(args ZapWrapperArgs) Wrapper {
+	if args.Level == NopLevel {
 		return NopWrapper
 	}
 
 	return func(ctx context.Context, msg string) {
 		logger := C(ctx)
 		// For unknown values, fallback to info level.
-		f := logger.Info
-		switch level {
+		f := logger.Infow
+		switch args.Level {
 		case DebugLevel:
-			f = logger.Debug
+			f = logger.Debugw
 		case WarnLevel:
-			f = logger.Warn
+			f = logger.Warnw
 		case ErrorLevel:
-			f = logger.Error
+			f = logger.Errorw
 		case PanicLevel:
-			f = logger.Panic
+			f = logger.Panicw
 		case FatalLevel:
-			f = logger.Fatal
+			f = logger.Fatalw
+		}
+		kv := make([]interface{}, 0, len(args.KVPairs)*2)
+		for k, v := range args.KVPairs {
+			kv = append(kv, k, v)
 		}
```

Comment:
Do this once in the `ZapWrapper`.

1. Huge performance improvement; now you're not leaking a ton of boxed values and slices
2. No need to worry about concurrent modification of the ZapWrapperArgs map.